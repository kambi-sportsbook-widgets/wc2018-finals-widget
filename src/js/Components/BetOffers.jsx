import React from 'react'
import PropTypes from 'prop-types'
import { OutcomeButton } from 'kambi-widget-components'
import styles from './BetOffers.scss'

const BetOffers = ({ event, betOffers, isMobile }) => {
  return (
    <div
      className={`${styles.betofferContainer} ${
        isMobile ? styles.betofferContainerMobile : ''
      }`}
    >
      {betOffers.map(betOffer => {
        if (betOffer.outcomes && betOffer.outcomes.length) {
          const mainClass = betOffer.main ? ' main' : ''
          return (
            <div
              className={`${styles.event} ${mainClass} ${
                isMobile ? styles.eventMobile : ''
              }`}
              key={betOffer.id}
            >
              <span className={styles.betofferLabel}>
                {betOffer.criterion.label}
              </span>
              <div className={styles.outcomes}>
                {betOffer.outcomes.map(outcome => (
                  <div className={styles.outcome} key={outcome.id}>
                    {/* Outcome button component
                                  https://github.com/kambi-sportsbook-widgets/widget-components/blob/master/README.md
                                 */}
                    <OutcomeButton
                      outcome={outcome}
                      outlineStyle={true}
                      event={event}
                      updateOdds={false}
                    />
                  </div>
                ))}
              </div>
            </div>
          )
        }
      })}
    </div>
  )
}

BetOffers.propTypes = {
  betOffers: PropTypes.array.isRequired,
  event: PropTypes.object.isRequired,
}

export default BetOffers
