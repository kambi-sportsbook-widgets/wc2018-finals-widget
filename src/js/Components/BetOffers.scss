.betoffer-container {
  flex-basis: 35%;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.event {
  display: flex;
  flex-direction: column;
  position: relative;
  margin-top: .7em;
}

.outcomes {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
}

.outcome {
  display: flex;
  flex-basis: auto;
  padding: 3px 5px;
  flex-grow: 1;

  :global(.KambiWidget-outcome__flexwrap) {
    justify-content: space-around;
  }

  :global(.KambiWidget-outcome__label-wrapper) {
    flex: none;
  }
}

.betoffer-label {
  text-align: left;
  padding-left: .5em;
}

@media screen and (max-width: 768px) {
  .betoffer-container {
    flex-basis: 100%;
  }

  .event:not(:global(.main)) {
    display: none;
  }
}
