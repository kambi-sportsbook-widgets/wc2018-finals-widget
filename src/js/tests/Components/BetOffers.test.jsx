// Don't load svg strings into tests
jest.mock('../../../../node_modules/kambi-widget-components/src/Carousel/chevron_left.svg', () => <svg>Placeholder</svg>)
jest.mock('../../../../node_modules/kambi-widget-components/src/Carousel/chevron_right.svg', () => <svg>Placeholder</svg>)

import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import BetOffers from '../../Components/BetOffers'

let renderer;

const mockEvent = {}

const emptyBetOffers = []

const exampleBetOffers = [
  {
    main: true,
    id: 1,
    criterion: { label: 'Some label 1' },
    outcomes: [ { id: 1 }, { id: 2 }, { id: 3 } ]
  },
  {
    id: 2,
    criterion: { label: 'Some label 2' },
    outcomes: [ { id: 4 }, { id: 5 }, { id: 6 } ]
  },
  {
    id: 3,
    criterion: { label: 'Some label 3' },
    outcomes: [ { id: 7 }, { id: 8 }, { id: 9 } ]
  },
]

describe('BetOffers DOM rendering', () => {

  beforeEach(() => {
    renderer = new ReactShallowRenderer()
  })

  it('renders correctly with empty props', () => {
    expect(
      renderer.render(
        <BetOffers
        betOffers={emptyBetOffers}
        event={mockEvent}
        />
      )
    )
    .toMatchSnapshot()
  })

  it('renders correctly with example props', () => {
    expect(
      renderer.render(
        <BetOffers
          betOffers={exampleBetOffers}
          event={mockEvent}
        />
      )
    )
    .toMatchSnapshot()
  })
})
