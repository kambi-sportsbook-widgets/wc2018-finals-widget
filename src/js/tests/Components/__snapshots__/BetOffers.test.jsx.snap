// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetOffers DOM rendering renders correctly with empty props 1`] = `
<div
  className="betoffer-container"
/>
`;

exports[`BetOffers DOM rendering renders correctly with example props 1`] = `
<div
  className="betoffer-container"
>
  <div
    className="event main"
  >
    <span
      className="betoffer-label"
    >
      Some label 1
    </span>
    <div
      className="outcomes"
    >
      <div
        className="outcome"
      >
        <OutcomeButton
          event={Object {}}
          label={true}
          outcome={
            Object {
              "id": 1,
            }
          }
          outlineStyle={true}
        />
      </div>
      <div
        className="outcome"
      >
        <OutcomeButton
          event={Object {}}
          label={true}
          outcome={
            Object {
              "id": 2,
            }
          }
          outlineStyle={true}
        />
      </div>
      <div
        className="outcome"
      >
        <OutcomeButton
          event={Object {}}
          label={true}
          outcome={
            Object {
              "id": 3,
            }
          }
          outlineStyle={true}
        />
      </div>
    </div>
  </div>
  <div
    className="event"
  >
    <span
      className="betoffer-label"
    >
      Some label 2
    </span>
    <div
      className="outcomes"
    >
      <div
        className="outcome"
      >
        <OutcomeButton
          event={Object {}}
          label={true}
          outcome={
            Object {
              "id": 4,
            }
          }
          outlineStyle={true}
        />
      </div>
      <div
        className="outcome"
      >
        <OutcomeButton
          event={Object {}}
          label={true}
          outcome={
            Object {
              "id": 5,
            }
          }
          outlineStyle={true}
        />
      </div>
      <div
        className="outcome"
      >
        <OutcomeButton
          event={Object {}}
          label={true}
          outcome={
            Object {
              "id": 6,
            }
          }
          outlineStyle={true}
        />
      </div>
    </div>
  </div>
  <div
    className="event"
  >
    <span
      className="betoffer-label"
    >
      Some label 3
    </span>
    <div
      className="outcomes"
    >
      <div
        className="outcome"
      >
        <OutcomeButton
          event={Object {}}
          label={true}
          outcome={
            Object {
              "id": 7,
            }
          }
          outlineStyle={true}
        />
      </div>
      <div
        className="outcome"
      >
        <OutcomeButton
          event={Object {}}
          label={true}
          outcome={
            Object {
              "id": 8,
            }
          }
          outlineStyle={true}
        />
      </div>
      <div
        className="outcome"
      >
        <OutcomeButton
          event={Object {}}
          label={true}
          outcome={
            Object {
              "id": 9,
            }
          }
          outlineStyle={true}
        />
      </div>
    </div>
  </div>
</div>
`;
